### PR TITLE
fix: bring back OTEL receiver metrics

### LINKED ISSUE #3758: Tempo (v2.5.0) Distributor missing tempo_receiver_accepted_spans
**Describe the bug**
Today I upgraded the tempo-distributed helm chart from 1.8.5 to 1.10.0 (app version 2.3.1 to [2.5.0](https://github.com/grafana/tempo/releases/tag/v2.5.0)) and I noticed in the Tempo Operational dashboard, that the `tempo_receiver_accepted_spans` metric disappeared. I'm assuming this the result to a change in the embedded OTLP library but other than the change noted in the release, there was nothing specific about this metric, making me think breaking it was unintentional.

**To Reproduce**
1. Install version 1.10.0 of the tempo-distributed helm chart.
2. Verify the absence of `tempo_receiver_accepted_spans` at /metrics of the tempo distributor component.

**Expected behavior**
I would expect the metric `tempo_receiver_accepted_spans` to still exist/be-emitted.

**Environment:**
 - Infrastructure: GKE fully managed
 - Deployment tool: Kustomize plus helm (`helmCharts`) [chart](https://github.com/grafana/helm-charts/tree/main/charts/tempo-distributed)

**Additional Context**
Here is a graph showing `tempo_receiver_` metrics disappearing while `tempo_distributor_spans_received_total` remains.
image

### PR DESCRIPTION
**Pull request by javiermolinar**
**What this PR does**:
This PR recovers the OTEL receiver metrics that Tempo previously exported.

The metrics stopped from being exported in this commit: #3148.
Doing a bit of code archeology here is where the OpenCensus code is removed https://github.com/open-telemetry/opentelemetry-collector/pull/9102.

The way this works is by implementing a custom MeterProvider, adapted from [go.opentelemetry.io/otel/metric/noop](go.opentelemetry.io/otel/metric/noop), to intercept the metrics emitted by obsreport.go and record them using our internal tooling.

Example of the metrics from the metrics endpoint:

```
# HELP tempo_receiver_accepted_spans Number of spans successfully pushed into the pipeline.
# TYPE tempo_receiver_accepted_spans counter
tempo_receiver_accepted_spans{receiver="tempo/otlp_receiver",transport="http"} 1
# HELP tempo_receiver_refused_spans Number of spans that could not be pushed into the pipeline.
# TYPE tempo_receiver_refused_spans counter
tempo_receiver_refused_spans{receiver="tempo/otlp_receiver",transport="http"} 0
```

**Which issue(s) this PR fixes**:
Fixes #3758

**Checklist**
- [X] Tests updated
- [ ] Documentation added
- [X] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: modules/distributor/receiver/metrics_provider.go [hunk 1]
```diff
@@ -0,0 +1,325 @@
...
+	return Meter{}
+}
+
+// Meter is an OpenTelemetry No-Op Meter.
+type Meter struct{ embedded.Meter }
+
+// Int64Counter returns a Counter used to record int64 measurements that
+// produces no telemetry.
+func (Meter) Int64Counter(name string, _ ...metric.Int64CounterOption) (metric.Int64Counter, error) {
+	return Int64Counter{Name: name}, nil
```
Comment (joe-elliott) [inline modules/distributor/receiver/metrics_provider.go:84]: if the name is not one we care to track we should just return a `NopInt64Counter`.

**Review by joe-elliott — COMMENTED** [review]
Nice fix! One small perf tweak and we're good.

File: modules/distributor/receiver/metrics_provider.go (hunk 1 above)
Comment (javiermolinar) [reply modules/distributor/receiver/metrics_provider.go:84]: We need it, without storing the name there is no way we can know what metric we are exporting:

https://github.com/grafana/tempo/blob/3798685b1fb78e576c2213f1f8115e4762d61725/modules/distributor/receiver/metrics_provider.go#L215

**Comment by zlin179** [comment]
Just wondering will we get a patch to 2.5 soon or we need to wait until 2.6 is released for this fix?

**Comment by joe-elliott** [comment]
> Just wondering will we get a patch to 2.5 soon or we need to wait until 2.6 is released for this fix?

This will be released with 2.6. This will likely occur towards the end of August

**Review by joe-elliott — APPROVED** [review]

**Comment by zlin179** [comment]
> > Just wondering will we get a patch to 2.5 soon or we need to wait until 2.6 is released for this fix?
> 
> This will be released with 2.6. The release will likely occur towards the end of August

Thanks. Then is there any suggestion/workaround on checking logs/metrics on refused spans while `receiver_refused_spans` is missing? The only metrics I see is `distributor_spans_received_total` but it only shows the number of accepted_spans right?

**Comment by joe-elliott** [comment]
> Then is there any suggestion/workaround on checking logs/metrics on refused spans while receiver_refused_spans is missing?

I would use: `tempo_discarded_spans_total`. It shows spans dropped for a number of reasons including ingestion reasons

**Pull request merged**